addons/sms/models/mail_notification.py

```diff
@@ -19,3 +20,38 @@ class MailNotification(models.Model):
         ('sms_server', 'Server Error'),
         ('sms_acc', 'Unregistered Account')
     ])
+
+    @api.model
+    def _cron_notify_admins(self):
+        super()._cron_notify_admins()
+        previous_date = fields.Datetime.now() - timedelta(days=1)
+        # Count failed SMS
+        failed_sms_counter = self.env['sms.sms'].sudo().search_count([
+            ('create_date', '>=', previous_date.strftime(tools.DEFAULT_SERVER_DATETIME_FORMAT)),
+            ('state', '=', 'error')])
+        sms_messages = {
+            "sms__failure": lambda failed_sms_counter, mail_sms_notifications_window_action: (
+                _('Odoo has been unable to send SMS'),
+                _("""%(failed_sms_counter)d SMS(s) could not be sent due to technical issues.<br/>
+                Please check whether recipient's phone number exists and if your have enaugh IAP credits.<br/>
+                To recharge your IAP credits please go to your "IAP Portal" <b>[Settings app > Odoo IAP > View my Services]</b>.<br/>
+                From there, you can view your current balance, recharge your credits, review your consumption and set another 
+                reminder (by email) to when credits are low.<br/>
+                You can check SMS <a href="/web#action=%(mail_sms_notifications_window_action)s&amp;view_type=list">notifications</a> for more details about failed SMS 
+                messages (an error message should help you diagnose the issue.)
+                """) % {
+                    'failed_sms_counter': failed_sms_counter,
+                    'mail_sms_notifications_window_action': mail_sms_notifications_window_action
+                },
+                'https://www.odoo.com/documentation/user/online/general/in_app_purchase/in_app_purchase.html'
+            )
+        }
+        # Send notifications about failed SMS (_notify_admins methods in daily cron job should set a repeat_delay of 1 day)
+        if failed_sms_counter:
+            self._notify_admins(
+                *self._get_admin_notification('sms__failure', sms_messages)(
+                    failed_sms_counter,
+                    self.env.ref('mail.mail_notification_action').id
+                ),
```

Comment:
I don't really get the point of calling a method where you give what you except to receive from that method.
Why not override the method to add the failure type and message, than call this method here ?
or even directly call _notify_admins(sms_message) (where you already fill the needed variables ? (first one is probably better --> see other related comment)